File: scripts/add-police-filing.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
批量添加公安备案信息到所有 HTML 页面
使用二进制方式读取和写入，避免编码问题
"""

import os

WEB_DIR = '/Users/jinkun/storytree/web'

def modify_file(filepath):
    """修改单个文件"""
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    # 检查是否已经包含公安备案
    if 'beian.mps.gov.cn' in content:
        print(f'⏭️  跳过 {os.path.basename(filepath)} - 已包含公安备案')
        return False

    # 旧的备案信息 HTML（普通格式）
    old_plain = '<a href="https://beian.miit.gov.cn/" target="_blank" rel="noopener noreferrer">浙 ICP 备 2026023946 号 -2</a>'

    # 旧的备案信息 HTML（带样式格式，用于 login.html, register.html 等）
    old_styled = '<a href="https://beian.miit.gov.cn/" target="_blank" rel="noopener noreferrer" style="margin-top: 6px; display: inline-block;">浙 ICP 备 2026023946 号 -2</a>'

    # 新的备案信息 HTML
    new_filing = '''<a href="https://beian.miit.gov.cn/" target="_blank" rel="noopener noreferrer" style="color: inherit; text-decoration: none;">浙 ICP 备 2026023946 号 -2</a>
            <span style="margin: 0 8px;">|</span>
            <a href="https://beian.mps.gov.cn/#/query/webSearch?code=33011002019543" rel="noreferrer" target="_blank" style="color: inherit; text-decoration: none; display: inline-flex; align-items: center; gap: 4px;">
                <img src="/assets/beian-icon.png" alt="公安备案图标" style="width: 14px; height: 14px; vertical-align: middle;">
                浙公网安备 33011002019543 号
            </a>'''

    modified = False

    # 尝试替换带样式的版本
    if old_styled in content:
        content = content.replace(old_styled, new_filing)
        modified = True
        print(f'✅ 修改 {os.path.basename(filepath)} (带样式)')

    # 尝试替换普通版本
    if old_plain in content:
        content = content.replace(old_plain, new_filing)
        modified = True
        print(f'✅ 修改 {os.path.basename(filepath)}')

    if modified:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content)

    return modified

def main():
    # 获取所有 HTML 文件
    html_files = [f for f in os.listdir(WEB_DIR) if f.endswith('.html')]

    print(f'📁 找到 {len(html_files)} 个 HTML 文件\n')

    modified_count = 0
    skipped_count = 0

    for filename in sorted(html_files):
        filepath = os.path.join(WEB_DIR, filename)
        try:
            if modify_file(filepath):
                modified_count += 1
            else:
                skipped_count += 1
        except Exception as e:
            print(f'❌ 错误 {filename}: {e}')

    print(f'\n📊 完成：修改了 {modified_count} 个文件，跳过了 {skipped_count} 个文件')
    print(f'\n💡 提示：请确保 /assets/beian-icon.png 文件存在')

if __name__ == '__main__':
    main()

File: scripts/add-police-filing.js
const fs = require('fs');
const path = require('path');

const WEB_DIR = '/Users/jinkun/storytree/web';

const NEW_FILING = `<a href="https://beian.miit.gov.cn/" target="_blank" rel="noopener noreferrer" style="color: inherit; text-decoration: none;">浙ICP 备 2026023946 号 -2</a>
            <span style="margin: 0 8px;">|</span>
            <a href="https://beian.mps.gov.cn/#/query/webSearch?code=33011002019543" rel="noreferrer" target="_blank" style="color: inherit; text-decoration: none; display: inline-flex; align-items: center; gap: 4px;">
                <img src="/assets/beian-icon.png" alt="公安备案图标" style="width: 14px; height: 14px; vertical-align: middle;">
                浙公网安备 33011002019543 号
            </a>`;

// 获取所有 HTML 文件
const files = fs.readdirSync(WEB_DIR).filter(f => f.endsWith('.html'));

console.log(`📁 找到 ${files.length} 个 HTML 文件\n`);

let modifiedCount = 0;
let skippedCount = 0;

files.forEach(filename => {
    const filepath = path.join(WEB_DIR, filename);
    try {
        let content = fs.readFileSync(filepath, 'utf-8');

        // 检查是否已经包含公安备案
        if (content.includes('beian.mps.gov.cn')) {
            console.log(`⏭️  跳过 ${filename} - 已包含公安备案`);
            skippedCount++;
            return;
        }

        // 查找 '浙 I' 来定位备案号
        let idx = -1;
        for (let i = 0; i < content.length; i++) {
            if (content[i] === '浙' && content[i+1] === 'I') {
                idx = i;
                break;
            }
        }

        if (idx < 0) {
            console.log(`⚠️  跳过 ${filename} - 未找到 ICP 备案`);
            skippedCount++;
            return;
        }

        // 提取完整标签
        const start = content.lastIndexOf('<a', idx);
        const end = content.indexOf('</a>', idx) + 4;
        const tag = content.substring(start, end);

        // 替换
        const newContent = content.replace(tag, NEW_FILING);

        if (newContent !== content) {
            fs.writeFileSync(filepath, newContent, 'utf-8');
            console.log(`✅ 修改 ${filename}`);
            modifiedCount++;
        } else {
            console.log(`⚠️  跳过 ${filename} - 替换失败`);
            skippedCount++;
        }
    } catch (e) {
        console.log(`❌ 错误 ${filename}: ${e.message}`);
    }
});

console.log(`\n📊 完成：修改了 ${modifiedCount} 个文件，跳过了 ${skippedCount} 个文件`);
console.log(`\n💡 提示：请确保 /assets/beian-icon.png 文件存在`);